### PR TITLE
blockchain: Add median time tests.

### PR DESCRIPTION
**Pull request by davecgh**
**This requires #990**.

This adds several tests for the `CalcPastMedianTime` function including when there are less than the usual number of blocks which happens near the beginning of the chain.

### CODE REVIEW

File: blockchain/blockindex.go [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+}
+
+// newBlockNode returns a new block node for the given block header.  It is
+// completely disconnected from the chain and the workSum value is just the work
+// for the passed block.  The work sum is updated accordingly when the node is
+// inserted into a chain.
+func newBlockNode(blockHeader *wire.BlockHeader, spentTickets *stake.SpentTicketsInBlock) *blockNode {
+	// Serialize the block header for use in calculating the initialization
+	// vector for the ticket lottery.  The only way this can fail is if the
+	// process is out of memory in which case it would panic anyways, so altough
```
Comment (dnldd) [inline blockchain/blockindex.go]: typo: "altough" -> although

File: blockchain/blockindex.go [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+
+	// Add the new node to the indices for faster lookups.
+	bi.index[*hash] = node
+	bi.depNodes[*prevHash] = append(bi.depNodes[*prevHash], node)
+
+	return node, nil
+}
+
+// PrevNodeFromBlock returns a block node for the block previous to the
+// passed block (the passed block's parent).  When it is already in the memory
```
Comment (dnldd) [inline blockchain/blockindex.go]: suggestion: `When it is already in the memory block chain` could be simplified to  `When it is already in the block index`

File: blockchain/blockindex.go (hunk 1 above)
Comment (davecgh) [reply blockchain/blockindex.go]: This is in #990.  Fixed there.

File: blockchain/blockindex.go (hunk 2 above)
Comment (davecgh) [reply blockchain/blockindex.go]: This is in #990.  Fixed there.

**Pull request merged**